lib/js/src/rpc/structs/VideoStreamingCapability.js

```diff
@@ -72,15 +84,24 @@ class VideoStreamingCapability extends RpcStruct {
     * @return {VideoStreamingCapability}
     */
     setSupportedFormats (val) {
-        this.setParameter(VideoStreamingCapability.KEY_SUPPORTED_FORMATS, val);
-        return this;
+        const errorMessage = 'val must be an array of VideoStreamingFormat instances';
+        if (Array.isArray(val)) {
+            for (const item of val) {
+                if (!(item instanceof VideoStreamingFormat)) {
+                    throw new Error(errorMessage);
+                }
+            }
+            this.setParameter(VideoStreamingCapability.KEY_SUPPORTED_FORMATS, val);
+            return this;
+        }
+        throw new Error(errorMessage);
```

Comment:
Validation should be done via the `validateType` method provided in the base `RpcStruct` class.